docs/proposals/improve-addons.md

```diff
@@ -0,0 +1,350 @@
+  - [OptionalAddon (old name: Accessible Addon)](#optionaladdon-old-name-accessible-addon)
+  - [CustomerAddon (old name: Custom Addon)](#customeraddon-old-name-custom-addon)
+- [Task & effort](#task--effort)
+
+## Introduction
+
+This proposal introduces improvements to the current [Addon system](https://docs.kubermatic.com/kubermatic/v2.18/guides/addons/). It aims to make current addons more flexible and user-friendly. Additionally, it simplifies the usage of addons we ship with (almost) every cluster, so-called "Default Addons".
+
+## Motivation and Background
+
+Currently, KKP only supports one mechanism for addons: For this, all addons are stored in a single docker image that is configured in KKP. Afterwards, an initContainer is being run that starts the image and copies all addons onto the seed-controller-managers local file-system. From there, addon manifests are being rendered and rolled out to user clusters.
+
+The current implementation of Addons has some flaws that make the usage and maintenance of Addons cumbersome for cluster administrators:
+
+- To add “Custom Addons” the cluster-admin needs to bundle manifests and templates for these Addons in a Docker image that is based on the official addon-manifest docker image by Kubermatic.
+- It’s currently only possible to deploy multiple instances of the same Addon when using a workaround
+- Addons are strictly tied to KKP versions due to the composition logic of the manifest image tag; for any provided manifest image the tag gets overwritten with `:<KKP VERSION>\[-CUSTOM SUFFIX\]`
+- Mandatory core components of user clusters are named “Default Addons” which implies that these can be deleted by Cluster Admins
+- Complex Addons can leave a trail of reconciliation errors until successfully deployed because all manifest files are applied at the same time
+- Creating a cluster is a two-step process; firstly you have to create a cluster and only  afterward you can select custom Addons
+
+*note: PS team report that some clients are using weekly KKP update in dev environments, consequently they need to rebuild the addon image every week which is painful and error-prone.*
+
+## Goals
+
+- Decouple custom addons from the KKP lifecycle
+- Handle dependency **inside** an Addon (e.g. CRD need to be created before CRs, namespace needs to be created before other objects)
+- Addons are correctly reconciled on changes. [kubermatic/kubermatic#6035](https://github.com/kubermatic/kubermatic/issues/6035)
+- Be able to choose which addons should be installed during the cluster creation or in the cluster template
+- Be compatible with private registries in preparation for future air-gapped environment stories. The current Policy regarding the air-gapped environment is to assume a private trusted registry is available. Addon Source (manifest of the workload to deployed) and workload images of the addon should be  configurable to point on a private registry
```

Comment:
Just an idea, but if we are improving the addons then perhaps we can solve issue like https://github.com/kubermatic/dashboard/issues/3666 too. So make k8s/dashboard an addon that will be always deployed but the version used will differ based on the user cluster version.